vllm/utils.py

```diff
@@ -616,23 +626,46 @@ def str_to_int_tuple(s: str) -> Tuple[int, ...]:
             f"(e.g., 1, 2, 3). Given input: {s}") from e
 
 
-def make_tensor_with_pad(
+def make_array_with_pad(
     x: List[List[int]],
     max_len: int,
     pad: int,
-    dtype: torch.dtype,
-    device: Optional[Union[str, torch.device]],
-) -> torch.Tensor:
-    """Make a padded tensor of a 2D inputs.
+    dtype: npt.DTypeLike,
+) -> npt.NDArray:
+    """
+    Make a padded array from 2D inputs.
 
     The padding is applied to the end of each inner list until it reaches
     `max_len`.
     """
-    padded_x = np.zeros([len(x), max_len], dtype=np.int32) + pad
+    padded_x = np.full((len(x), max_len), pad, dtype=dtype)
```

Comment:
`np.full(..., pad)` is more efficient than `np.zeros(...) + pad`. Try it yourself:

```bash
python -m timeit "import numpy as np; np.zeros(100000) + 2"
python -m timeit "import numpy as np; np.full(100000, 2)"
```

Reply:
I've also fixed the dtype to be consistent with the pytorch one.